demo/apisix/conf/apisix.yaml

```diff
@@ -6,6 +6,82 @@ routes:
       nodes:
       type: roundrobin
     plugin_config_id: 1
+  -
+    id: jmap
+    uri: /jmap
+    service_id: jmap_service
+    methods:
+      - POST
+  -
+    id: jmap_session
+    uri: /jmap/session
+    service_id: jmap_service
+    methods:
+      - GET
+  -
+    id: download
+    uri: /download/*
+    service_id: jmap_service
+    methods:
+      - GET
+      - OPTIONS
+  -
+    id: upload
+    uri: /upload
+    service_id: jmap_service
+    methods:
+      - POST
+      - OPTIONS
+  -
+    id: upload
+    uri: /upload/*
+    service_id: jmap_service
+    methods:
+      - POST
+      - OPTIONS
+  -
+    id: web_known_finger
+    uri: /.well-known/webfinger
+    service_id: tmail_no_auth_service
+    methods:
+      - GET
+  -
+    id: web_known_jmap
+    uri: /.well-known/jmap
+    service_id: jmap_service
+    methods:
+      - GET
+  -
+    id: web_known_linagora_ecosystem
+    uri: /.well-known/linagora-ecosystem
+    service_id: jmap_service
+    methods:
+      - GET
+
+
+services:
+  -
+    id: jmap_service
+    upstream_id: tmail_upstream
+    plugins:
+      openid-connect:
+        client_id: "james_apisix"
+        client_secret: "james_apisix"
+        discovery: "http://sso.example.com/.well-known/openid-configuration"
+        scope: "openid profile"
+        bearer_only: true
+        introspection_endpoint_auth_method: "client_secret_post"
+        redirect_uri: "http://test.sso.example.com:8080/login-callback.html"
+  -
+    id: tmail_no_auth_service
+    upstream_id: tmail_upstream
+
+upstreams:
+  -
+    id: tmail_upstream
```

Comment:
`jmap_upstream` or something like that would be more clear IMO.

TMail can have many upstream: jmap, webadmin (not now but just say), websocket...